dfget/core/downloader/p2p_downloader/p2p_downloader.go

```diff
@@ -251,8 +252,9 @@ func (p2p *P2PDownloader) pullPieceTask(item *Piece) (
 
 	logrus.Errorf("pull piece task fail:%v and will migrate", res)
 	var registerRes *regist.RegisterResult
-	if registerRes, err = p2p.Register.Register(p2p.cfg.RV.PeerPort); err != nil {
-		return nil, err
+	var registerErr *errortypes.DfError
```

Comment:
No need to define `registerErr` here if we will not use it out of the  block 
```
if registerRes, registerErr = p2p.Register.Register(p2p.cfg.RV.PeerPort)
```

Reply:
registerErr is *errortypes.DfError type, if we re-use variable err here, it is always no-nil. See golangci-lint output below.
```
dfget/core/downloader/p2p_downloader/p2p_downloader.go:255:72: nilness: tautological condition: non-nil != nil (govet)
        if registerRes, err = p2p.Register.Register(p2p.cfg.RV.PeerPort); err != nil {
```

Reply:
```
registerRes, registerErr := p2p.Register.Register(p2p.cfg.RV.PeerPort)
if registerErr != nil {
	return nil, registerErr
}
```